src/util.cpp

```diff
@@ -488,6 +488,12 @@ namespace Sass {
 
       int cp = utf8::next(it, end);
 
+      // in case of \r, check if the next in sequence
+      // is \n and then advance the iterator.
+      if (cp == '\r' && it < end && utf8::peek_next(it, end) == '\n') {
```

Comment:
Looking through the rest of this file it looks like we generally treat `\r` as a noop. I think you can drop the peek here.